Index: build/gyp/pylib/gyp/xcode_emulation.py
===================================================================

--- build/gyp/pylib/gyp/xcode_emulation.py	2014-12-03 15:51:45.000000000 -0800
+++ build/gyp/pylib/gyp/xcode_emulation.py	2014-12-03 15:53:43.000000000 -0800
@@ -295,9 +295,6 @@ class XcodeSettings(object):
     if self._Test('GCC_SYMBOLS_PRIVATE_EXTERN', 'YES', default='NO'):
       cflags.append('-fvisibility=hidden')
 
-    if self._Test('GCC_TREAT_WARNINGS_AS_ERRORS', 'YES', default='NO'):
-      cflags.append('-Werror')
-
     if self._Test('GCC_WARN_ABOUT_MISSING_NEWLINE', 'YES', default='NO'):
       cflags.append('-Wnewline-eof')               